src/plugins/data/public/ui/typeahead/suggestions_component.tsx

```diff
@@ -142,14 +144,130 @@ export default class SuggestionsComponent extends Component<SuggestionsComponent
     if (remaining <= margin) {
       this.props.loadMore();
     }
-  };
+  });
 }
 
 const StyledSuggestionsListDiv = styled.div`
-  ${(props: { queryBarRect: DOMRect; verticalListPosition: string }) => `
+  ${(props: { left: number; width: number; verticalListPosition: string }) => `
       position: absolute;
       z-index: 4001;
-      left: ${props.queryBarRect.left}px;
-      width: ${props.queryBarRect.width}px;
+      left: ${props.left}px;
+      width: ${props.width}px;
       ${props.verticalListPosition}`}
 `;
+
+const ResizableSuggestionsListDiv: React.FC<{
+  inputContainer: HTMLElement;
+  suggestionsSize?: SuggestionsListSize;
+}> = React.memo((props) => {
+  const inputContainer = props.inputContainer;
+  const children = props.children as (rect: DOMRect) => ReactNode;
+
+  const [{ documentHeight }, { pageYOffset }, containerRect] = useDimensions(inputContainer);
+
+  if (!containerRect) return null;
+
+  // reflects if the suggestions list has enough space below to be opened down
+  const isSuggestionsListFittable =
+    documentHeight - (containerRect.top + containerRect.height) >
+    SUGGESTIONS_LIST_REQUIRED_BOTTOM_SPACE;
+  const verticalListPosition = isSuggestionsListFittable
+    ? `top: ${pageYOffset + containerRect.bottom - SUGGESTIONS_LIST_REQUIRED_TOP_OFFSET}px;`
+    : `bottom: ${documentHeight - (pageYOffset + containerRect.top)}px;`;
+
+  return (
+    <StyledSuggestionsListDiv
+      left={containerRect.left}
+      width={containerRect.width}
+      verticalListPosition={verticalListPosition}
+    >
+      <div
+        className={classNames('kbnTypeahead', {
+          'kbnTypeahead--small': props.suggestionsSize === 's',
+        })}
+      >
+        <div
+          className={classNames('kbnTypeahead__popover', {
+            ['kbnTypeahead__popover--bottom']: isSuggestionsListFittable,
+            ['kbnTypeahead__popover--top']: !isSuggestionsListFittable,
+          })}
+        >
+          {children(containerRect)}
+        </div>
+      </div>
+    </StyledSuggestionsListDiv>
+  );
+});
+
+function useDimensions(
```

Comment:
This is used for positioning suggestions component relative to the input field. Previously it was scattered between parent and child, now it is centralized inside a child. 

This allowed to avoid trigger parent's re-rendering on page resizes.